vpr/src/place/median_move_generator.cpp

```diff
@@ -0,0 +1,339 @@
+ * bb_coord_new data structure.  This routine should only be called   *
+ * for small nets, since it does not determine enough information for *
+ * the bounding box to be updated incrementally later.                *
+ * Currently assumes channels on both sides of the CLBs forming the   *
+ * edges of the bounding box can be used.  Essentially, I am assuming *
+ * the pins always lie on the outside of the bounding box.            */
+static void get_bb_from_scratch(ClusterNetId net_id, t_bb* bb_coord_new, ClusterBlockId block_id, bool& skip_net) {
+    //TODO: account for multiple physical pin instances per logical pin
+
+    skip_net = true;
+
+    int xmin = 0;
+    int xmax = 0;
+    int ymin = 0;
+    int ymax = 0;
+
+    int x, y;
+    int pnum;
+
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.placement();
+    auto& device_ctx = g_vpr_ctx.device();
+
+    ClusterBlockId bnum = cluster_ctx.clb_nlist.net_driver_block(net_id);
+    bool first_block = false;
+
+    if (bnum != block_id) {
+        skip_net = false;
+        pnum = net_pin_to_tile_pin_index(net_id, 0);
+        x = place_ctx.block_locs[bnum].loc.x + physical_tile_type(bnum)->pin_width_offset[pnum];
+        y = place_ctx.block_locs[bnum].loc.y + physical_tile_type(bnum)->pin_height_offset[pnum];
+
+        xmin = x;
+        ymin = y;
+        xmax = x;
+        ymax = y;
+
+        first_block = true;
+    }
+
+    for (auto pin_id : cluster_ctx.clb_nlist.net_sinks(net_id)) {
+        bnum = cluster_ctx.clb_nlist.pin_block(pin_id);
+        pnum = tile_pin_index(pin_id);
+        if (bnum == block_id)
+            continue;
+        skip_net = false;
+        x = place_ctx.block_locs[bnum].loc.x + physical_tile_type(bnum)->pin_width_offset[pnum];
+        y = place_ctx.block_locs[bnum].loc.y + physical_tile_type(bnum)->pin_height_offset[pnum];
+
+        if (!first_block) {
+            xmin = x;
+            ymin = y;
+            xmax = x;
+            ymax = y;
+            first_block = true;
+        }
+        if (x < xmin) {
+            xmin = x;
+        } else if (x > xmax) {
+            xmax = x;
+        }
+
+        if (y < ymin) {
+            ymin = y;
+        } else if (y > ymax) {
+            ymax = y;
+        }
+    }
+
+    /* Now I've found the coordinates of the bounding box.  There are no *
+     * channels beyond device_ctx.grid.width()-2 and                     *
+     * device_ctx.grid.height() - 2, so I want to clip to that.  As well,*
+     * since I'll always include the channel immediately below and the   *
+     * channel immediately to the left of the bounding box, I want to    *
+     * clip to 1 in both directions as well (since minimum channel index *
+     * is 0).  See route_common.cpp for a channel diagram.               */
+
+    bb_coord_new->xmin = std::max(std::min<int>(xmin, device_ctx.grid.width() - 2), 1);  //-2 for no perim channels
```

Comment:
Not sure why we're clipping to 1, width-1.  Should try changing1 the original code if this is copied from it.
This code all looks duplicated though so we should get rid of the duplicate.

Reply:
Opened an issue to solve this in a separate PR as it is originally used in the master branch as it is. Please track it on #1599